### ISSUE TITLE
Pod install fails for versions after 1.0.3 due to VersionChecker.swift

### ISSUE THREAD

**Issue by yiyingzz** [apollographql/apollo-ios#2688]
## Bug report

Installing Apollo via Cocoapods for versions after 1.0.4+ fails due to an issue with a file `VersionChecker.swift`. I checked the commits and it looks like this file was added 8 days ago before v1.0.4 was released. Installing a previous ver worked fine.

This is the error message:

```
/Users/y/Library/Caches/CocoaPods/Pods/Release/Apollo/1.0.4-7c977/Sources/CodegenCLI/Extensions/VersionChecker.swift:34:29: error: variable binding in a condition requires an initializer
      if let projectRootURL {
                            ^
/Users/y/Library/Caches/CocoaPods/Pods/Release/Apollo/1.0.4-7c977/Sources/CodegenCLI/Extensions/VersionChecker.swift:52:29: error: variable binding in a condition requires an initializer
      if let projectRootURL {
                            ^
make: *** [build-cli-for-cocoapods] Error 1
```

## Versions

Please fill in the versions you're currently using: 

- `apollo-ios` SDK version: 1.0.4+
- Xcode version: 13.4
- Swift version: 5.6
- Package manager: Cocoapods

## Steps to reproduce

Install Apollo 1.0.4 or 1.0.5 using Cocoapods

**Comment by calvincestari** [comment]
Hi @yiyingzz - `if let projectRootURL {` is the new `if let` shorthand from [SE-0435](https://github.com/apple/swift-evolution/blob/main/proposals/0345-if-let-shorthand.md). From that proposal you can see that it was implemented in Swift 5.7, which is [available in Xcode 14.0/14.1](https://developer.apple.com/support/xcode/). If you update your version of Xcode the compiler error will go away.

Is there a specific reason you won't be able to update Xcode?

**Comment by yiyingzz** [comment]
Hi @calvincestari - thanks for the quick response! I will have to check with our team if we can update it as we were on 13.x due to some build issues.
